src/unitxt/metrics.py

```diff
@@ -399,31 +531,48 @@ def process(self, stream: Stream, stream_name: Optional[str] = None) -> Generato
 
             instances.append(instance)
 
-        for reduction, fields in self.reduction_map.items():
-            assert (
-                reduction in self.implemented_reductions
-            ), f"Reduction {reduction} is not implemented, use one of {self.implemented_reductions}"
+        return instances, global_score
 
-            if reduction == "mean":
-                from statistics import mean
+    @staticmethod
+    def aggregate(instances, field_name):
+        scores = [instance["score"]["instance"][field_name] for instance in instances]
+        import warnings
 
-                for field_name in fields:
-                    scores = [
-                        instance["score"]["instance"][field_name]
-                        for instance in instances
-                    ]
-                    global_score[field_name] = mean(scores)
-                    if field_name == self.main_score:
-                        global_score["score"] = global_score[field_name]
-                        global_score["score_name"] = self.main_score
+        with warnings.catch_warnings():
+            # in case instances is empty, return NaN but avoid printing a RuntimeWarning
+            warnings.simplefilter("ignore", category=RuntimeWarning)
+            return np.nanmean(scores)
```

Comment:
what happened before your changes if some of the instance scores were NaN? Is it still the same behaviour?